opencog/moses/main/populate_atomspace.cc

```diff
@@ -33,8 +33,10 @@ void populate(AtomSpace *as, const CTable &ctable)
 	for (int j = 0; j < labels.size(); ++j) {
 		const vertex_seq &col = ctable.get_input_col_data(j);
 		type_node col_type = get_type_node(types[j]);
+		std::string arg1 = "$"+std::to_string(j+1);
+
 		const Handle &feature = createNode(col_type == id::boolean_type ? PREDICATE_NODE : SCHEMA_NODE,
-		                                   labels[j]);
+		                                   "$"+std::to_string(j+1));
```

Comment:
Why don't you use labels?

Reply:
The way the features should be stored in the atomspace should  be consistent with the way combo atomese converter converts combo trees to a program. Otherwise `add_atom` will not find the added atom from the atomspace. This was part of the big problem that caused the output inconsistency.

Reply:
But the converter can take `labels` in input, so can't labels be used all the way?

Reply:
Are you saying to pass to the converter the labels of the table ?

Reply:
@ngeiswei the converter takes labels but there are some instances where we might want to regenerate the variables from their arity, rather than keeping track of the labelled vars. such as in the representation.

Reply:
Oh, I see what you mean @kasimebrahim , indeed in combo there are no labels internally, unlike atomese... But if understand well Eman fixed in #117.